src/super_gradients/training/datasets/segmentation_datasets/pascal_voc_segmentation.py

```diff
@@ -96,3 +118,49 @@ def _get_pascal_labels(self):
                 [0, 64, 128],
             ]
         )
+
+
+class PascalAUG2012SegmentationDataSet(PascalVOC2012SegmentationDataSet):
+    """
+    PascalAUG2012SegmentationDataSet - Segmentation Data Set Class for Pascal AUG 2012 Data Set
+    """
+
+    def __init__(self, *args, **kwargs):
+        self.sample_suffix = '.jpg'
+        self.target_suffix = '.mat'
+        super().__init__(sample_suffix=self.sample_suffix, target_suffix=self.target_suffix, *args, **kwargs)
+
+    @staticmethod
+    def target_loader(target_path: str) -> Image:
+        """
+        target_loader
+            :param target_path: The path to the target data
+            :return:            The loaded target
+        """
+        mat = scipy.io.loadmat(target_path, mat_dtype=True, squeeze_me=True,
+                               struct_as_record=False)
+        mask = mat['GTcls'].Segmentation
+        return Image.fromarray(mask)
+
+
+class PascalVOCAndAUGUnifiedDataset(ConcatDataset):
+    """
+    Pascal VOC + AUG train dataset, aka `SBD` dataset contributed in "Semantic contours from inverse detectors".
+    This is class implement the common usage of the SBD and PascalVOC datasets as a unified augmented trainset.
+    The unified dataset includes a total of 10,582 samples and don't contains duplicate samples from the PascalVOC
+    validation set.
+    """
+    def __init__(self, **kwargs):
+        print(kwargs)
+        if any([kwargs.pop("list_file"), kwargs.pop("samples_sub_directory"), kwargs.pop("targets_sub_directory")]):
+            logger.warning("[list_file, samples_sub_directory, targets_sub_directory] arguments passed will not be used"
+                           " when passed to `PascalVOCAndAUGUnifiedDataset`. Those values are predefined for initiating"
+                           " the Pascal VOC + AUG training set.")
+        super().__init__(datasets=[
+            PascalVOC2012SegmentationDataSet(list_file="VOCdevkit/VOC2012/ImageSets/Segmentation/train.txt",
```

Comment:
Those arguments are hardcoded which is not a clean way to define those arguments. On the other hand this the proper way to use this class in the first place, and a different configuration will be semantically wrong (Not `PascalVOCAndAUGUnifiedDataset`). Similar approach can be found in `PascalVOCUnifiedDetectionTrainDataset` where `train_dataset_names = ["train2007", "val2007", "train2012", "val2012"]` are hardcoded.
If we do want the flexibility to change those parameters I suggest to pass dataset_params as follows:
```
train_dataset_params:
  voc_subset_params:
    list_file: 
    samples_sub_directory: 
    targets_sub_directory: 
  aug_subset_params:
    list_file: 
    samples_sub_directory: 
    targets_sub_directory: 
...
``` 
WDYT? @shaydeci @ofrimasad

Reply:
let's keep it hardcoded for now. I agree it is a unique case where these params are not expected to change